samples/nrf9160/cloud_client/src/main.c

```diff
@@ -286,6 +316,8 @@ void main(void)
 	if (err) {
 		LOG_ERR("Cloud backend could not be initialized, error: %d",
 			err);
+		LOG_INF("Rebooting");
```

Comment:
is rebooting useful here?

Reply:
Yes.  When I tested modem FOTA in this app, it hung during the phase of modem FOTA updates, controlled by the modem firmware, which prevent LTE connections.  Adding this reboot made it continue and finish.

Reply:
Then something in nrf_cloud_fota needs fixed I think

Reply:
Maybe?  Here is where this happens.  There are multiple phases:
1) download and reboot:
```
I: Downloaded 13768/13768 bytes (100%)
I: Download complete
I: Scheduling modem firmware upgrade at next boot
I: Download complete
I: CLOUD_EVT_FOTA_DONE
I: FOTA type: 1
+CEREG: 0
I: Active LTE mode changed: None
+CSCON: 0
I: Rebooting to complete FOTA update
*** Booting Zephyr OS build v2.6.0-rc1-ncs1-2275-g2716fad989a4  ***
```
2) MCUboot does something...
```
: Starting bootloader
I: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Swap type: none
I: Bootloader chainload address offset: 0x10000
I: Jumping to the first image slot
```
3) Boots again.  Here is where it fails to connect, because the modem is being updated I think.
```
E: Failed to open AT socket (err:-117)
I: Cloud client has started
E: Failed to send AT command (err:9)
E: Failed to obtain IMEI, error: -9
E: Cloud backend could not be initialized, error: -9
I: Rebooting
*** Booting Zephyr OS build v2.6.0-rc1-ncs1-2275-g2716fad989a4  ***
```
4) MCUboot does something again...
```
I: Starting bootloader
I: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Swap type: none
I: Bootloader chainload address offset: 0x10000
I: Jumping to the first image slot
```
5) Boots yet again.
```
I: Cloud client has started
I: Rebooting to complete modem FOTA
*** Booting Zephyr OS build v2.6.0-rc1-ncs1-2275-g2716fad989a4  ***
```
6) MCUboot does yet something else...
```
I: Starting bootloader
I: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
I: Secondary image: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Swap type: none
I: Bootloader chainload address offset: 0x10000
I: Jumping to the first image slot
```
7) finally boots for good
```
: Cloud client has started
I: FOTA updated modem
I: Connecting to LTE network, this may take several minutes...
+CEREG: 2,"900A","0509260F",7
I: Active LTE mode changed: LTE-M
+

Reply:
https://github.com/nrfconnect/sdk-nrf/pull/5551
😅